test/matrix-test.jl

```diff
@@ -59,3 +59,34 @@ end
   @test is_zero(reduce_mod(A, B))
   @test is_zero(reduce_mod(A, 2*B))
 end
+
+@testset "Eigenvalues of ZZMatrix" begin
+  A4 = matrix(ZZ,4,4, [6,-8,9,5,  5,-6,5,0,  0,6,5,4,  -6,8,-1,7]);
+  E4 = eigenvalues(A4);
+  @test length(E4) == 2;
+  @test E4 == [2,4] || E4 == [4,2]
+  E4m = eigenvalues_with_multiplicities(A4);
+  @test length(E4m) == 2;
+  @test E4m[1][2] == 1 || E4m[2][2] == 1
```

Comment:
```suggestion
  @test issetequal(E4m, [(2, 1), (4, 1)])
```
if I looked right at the matrix, this is what we expect from the tests here, right? If so, I think this change would make the test easier to read

Reply:
even without changing this to a `issetequal`, I think you wanna test that both multiplicities are 1 here, don't you?
```suggestion
  @test E4m[1][2] == 1
  @test E4m[2][2] == 1
```

Reply:
Thanks!  You're right: I did mean `&&`.  Anyway I have just replaced it with `issetequal` which I think is easier to read and comprehend -- the reader knows immediately what the expected answer is.  Already pushed.  I was racing to be ready for an appointment at 16:00, so made a couple of mistakes (sigh).